src/components/movieListWrapper/MovieListWrapper.js

```diff
@@ -0,0 +1,108 @@
+import { fetchPopularMovieList, fetchSearchMovieList } from '../../apis/fetchData';
+import { LAST_PAGE } from '../../constants/constant';
+import { completeMovieList, loadMovieList } from '../../domain/MovieService';
+
+class MovieListWrapper {
+  #currentPage
+  #title
+  #getName
+  #inputValue
+
+  constructor(title, getName, inputValue = '') {
+    this.#currentPage = 1;
+    this.#title = title;
+    this.#getName = getName;
+    this.#inputValue = inputValue;
+  }
+
+  async create() {
+    const section = document.querySelector('.item-view');
+    if (!section) return;
+    section.replaceChildren();
+
+    const title = document.createElement('h2');
+    title.textContent = this.#title;
+
+    const ul = document.createElement('ul');
+    ul.className = 'item-list';
+    ul.replaceChildren();
+
+    const addButton = document.createElement('button');
+    addButton.className = 'btn primary full-width';
+    addButton.textContent = '더 보기';
+
+    section.append(title, ul);
+
+    await this.updateMovieList(addButton);
+
+    section.appendChild(addButton);
+
+    addButton.addEventListener('click', async () => {
+      await this.updateMovieList(addButton);
+    });
+  }
+
+  plusCurrentPage() {
+    this.#currentPage += 1;
+  }
+
+  hasNextPage(lastPage) {
+    if (this.#currentPage === (lastPage < LAST_PAGE ? lastPage : LAST_PAGE)) {
```

Comment:
```suggestion
    if (this.#currentPage === lastPage) {
```
이렇게 해줘도 동일 할 것 같아요ㅎㅎ

Reply:
이 함수는 삭제되었지만 질문 드려도 될까요? 

lastPage는 api에서 받는 total_pages 값입니다. 이 값이 500이 넘어갈 경우 에러가 뜨는데 '영화 검색 api'에서는 이를 확인할 방법이 없어서 일단 안전하게 저렇게 처리해두었습니다. 
사이트에서 api에 관련된 정보를 찾아보아도 찾을 수 없었는데 이 부분은 나중에 에러를 띄우면 될까요 미리 처리하는 게 나을까요??🤗

Reply:
아하 그런 목적이었군요!

일단, 이런 경우는 프론트엔드에서 미리 방어를 쳐주기보다는 그냥 백엔드에서 에러가 나면 그 때 유저에게  해당 에러를 인지시켜주는 편인 것 같아요. (대신 에러를 인지시켜주는 방법은 텐텐이 만들어주신 토스트 처럼 전체 UI를 헤치지 않는 방법으로요ㅎㅎ)

`totalPage값이 500이 넘어갈 경우는 에러가 난다`와 같은 요건을 처리해주는건 프론트가 도메인 지식을 너무 깊이 알고있는것이기도 하고요ㅎㅎ 프론트는 그저 api 에러에 대한 후속처리를 잘 해주면 되겠죠?